src/main/java/com/conveyal/taui/controllers/WorkerController.java

```diff
@@ -206,14 +213,49 @@ private String jsonResponse (Response response, int statusCode, Object object) {
      * Fetch status of all unfinished jobs as a JSON list.
      */
     private String getAllJobs(Request request, Response response) {
-        return jsonResponse(response, HttpStatus.OK_200, broker.getJobSummary());
+        String accessGroup = request.attribute("accessGroup");
+        if (!AnalysisServerConfig.adminAccessGroup.equals(accessGroup)) {
+            response.status(401);
+            return "You do not have access.";
+        }
+
+        Collection<JobStatus> jobStatuses = broker.getJobSummary();
+        for (JobStatus jobStatus : jobStatuses) {
+            if (!jobStatus.jobId.equals("SUM")) {
+                jobStatus.regionalAnalysis = Persistence.regionalAnalyses
+                        .find(QueryBuilder.start("_id").is(jobStatus.jobId).get()).next();
+            }
+        }
+
+        return jsonResponse(response, HttpStatus.OK_200, jobStatuses);
     }
 
     /**
      * Report all workers that have recently contacted the broker as JSON list.
      */
     private String getAllWorkers(Request request, Response response) {
-        return jsonResponse(response, HttpStatus.OK_200, broker.getWorkerObservations());
+        String accessGroup = request.attribute("accessGroup");
+        if (!AnalysisServerConfig.adminAccessGroup.equals(accessGroup)) {
+            response.status(401);
+            return "You do not have access.";
+        }
+
+        Collection<WorkerObservation> observations = broker.getWorkerObservations();
+        Map<String, Bundle> bundleForNetworkId = new HashMap<>();
+        for (WorkerObservation observation : observations) {
+            List<Bundle> bundles = new ArrayList<>();
+            for (String networkId : observation.status.networks) {
+                Bundle bundle = bundleForNetworkId.get(networkId);
+                if (bundle == null) {
+                    bundle = Persistence.bundles.find(QueryBuilder.start("_id").is(networkId).get()).next();
+                    bundleForNetworkId.put(networkId, bundle);
+                }
+                bundles.add(bundle);
```

Comment:
It seems strange to me that all of the fields of a newly constructed WorkerObservation are filled in by code within the WorkerObservation class itself, but this one field is filled in within an HTTP handler method. Though I can see not wanting to put it in the constructor it might be better if it was an instance method, which could even be called within broker.getWorkerObservations, to cleanly separate the purpose of the methods.